src/main/java/net/rubygrapefruit/platform/internal/jni/WindowsFileEventFunctions.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2012 Adam Murdoch
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package net.rubygrapefruit.platform.internal.jni;
+
+import net.rubygrapefruit.platform.NativeException;
+import net.rubygrapefruit.platform.NativeIntegration;
+import net.rubygrapefruit.platform.file.FileWatcher;
+import net.rubygrapefruit.platform.file.FileWatcherCallback;
+import net.rubygrapefruit.platform.internal.FunctionResult;
+
+import java.util.Collection;
+
+public class WindowsFileEventFunctions implements NativeIntegration {
```

Comment:
It would be good to remove the duplication. This is basically exactly the same as the macOS implementation, with the difference being the body of the two native methods.

Reply:
Extracted.